yarn/src/main/scala/org/apache/spark/deploy/yarn/ExecutorDelegationTokenUpdater.scala

```diff
@@ -76,7 +76,8 @@ private[spark] class ExecutorDelegationTokenUpdater(
         SparkHadoopUtil.get.getTimeFromNowToRenewal(
           sparkConf, 0.8, UserGroupInformation.getCurrentUser.getCredentials)
       if (timeFromNowToRenewal <= 0) {
-        executorUpdaterRunnable.run()
+        // Wait a minutes to avoid cycle calling this method.
```

Comment:
Can you change this comment to be a bit more clear something like:

We just checked for new credentials but none were there, wait a minute and retry.  This handles the shutdown case where the staging directory may have been removed.